lib/bolt/cli.rb

```diff
@@ -787,11 +787,7 @@ def generate_types
     # Initializes a specified directory as a Bolt project and installs any modules
     # specified by the user, along with their dependencies
     def initialize_project
-      # Dir.pwd will return backslashes on Windows, but Pathname always uses
-      # forward slashes to concatenate paths. This results in paths like
-      # C:\User\Administrator/modules, which fail module install. This ensure
-      # forward slashes in the cwd path.
-      dir = File.expand_path(Dir.pwd)
+      dir = Dir.pwd
```

Comment:
@GabrielNagy Sorry just coming back to this to prep for Puppet 7 release. I'm not sure I follow why this change is needed? I think it may be causing the failure in the Windows tests in GH actions.

Reply:
`Dir.pwd` is monkey-patched by the `win32/dir` gem to return a backslash-separated path. Even though the gem is not required here, it could be that it's required through puppet 6 here https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/windows.rb#L30.

With puppet 7 `Dir.pwd` uses forward slashes, as we removed all references to `win32/dir`, so this workaround here is not needed anymore.

I assumed the bolt version is tightly coupled with the puppet version due to the dependency being somewhat strict (https://github.com/puppetlabs/bolt/blob/main/bolt.gemspec#L51). After bumping puppet to 7.0.0 in the gemspec, PR tests should no longer fail.

I can revert this hunk so the code will work with both puppet 6 and 7. Let me know your thoughts.

Reply:
Hm, it sounds like we aren't planning to pick up Puppet 7 in Bolt proper until our next major version, which we're planning for early next year. Once we pick up Puppet 7 though we won't need to support puppet 6, since we ship the Puppet that Bolt uses with Bolt. Long way of saying we should leave it, sorry for the hassle!

Reply:
No worries, left it in and rebased. Seems that PR tests are passing now, so this should be good to merge.